### ISSUE TITLE
Custom secrets engine path in the Vault provider

### ISSUE THREAD

**Issue by palourde** [sensu/sensu-docs#2682]
### Description
We will add support for custom secrets engine path in Vault via https://github.com/sensu/sensu-enterprise-go/pull/1222.

Therefore, we need to update the documentation to reflect that change.

### Category

This is an issue with:

- [ ] errata (error in doc content)
- [ ] new doc (requires a completely new doc)
- [x] update (add missing or refresh existing content)
- [ ] improvement (improve docs content or format)

### Affected pages
- https://docs.sensu.io/sensu-go/latest/reference/secrets

### Expected behavior
We probably need to introduce a second format for the `id` of a secret.

We currently use this example, `"id": "secret/ansible#token"`, where we always assumed the secrets engine was mounted as `secret/` because this is the default value in all the Vault docs.

However, the [KV secrets engine v2](https://www.vaultproject.io/docs/secrets/kv) allows you to mount engines in different paths. Because of that, we needed to introduce a new delimiter (`#`) in the ID to allow users to explicitly specify that. Therefore, the following _id_ is now valid when using the v2 Vault KV version (see the `version` attribute in https://docs.sensu.io/sensu-go/latest/reference/secrets-providers/#client-attributes).:

```
secrets/sensu#ansible#token
```

Where:
- `secrets/sensu` corresponds to the secrets engine path
- `ansible` the path to the secret within that secrets engine
- `token` the field within that secret

The change we are introducing is backward compatible, therefore the previous format (`secret/ansible#token`) will continue to work with a v2 KV. However, I believe we should only document this new format for operators of v2 KV, simply because it's always better to explicitly define it.

#### In Summary

In our documentation of the `id` attribute](https://docs.sensu.io/sensu-go/latest/reference/secrets/#spec-attributes), we should introduce a distinction depending on whether the users are operating a v1 or v2 KV secrets engine.

- If v1, keep it the same: `"id": "secret/ansible#token"`
- If v2, document this new format: `"id": "secrets/sensu#ansible#token"`

I realize this is super confusing, even for me, so feel free to poke me if anything is unclear!

### Additional context

* Browser(s) and version(s): 
* OS:
